lib/pos_remote_node.go

```diff
@@ -0,0 +1,456 @@
+package lib
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/deso-protocol/core/bls"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/sha3"
+	"math"
+	"net"
+	"time"
+)
+
+type RNConnectionType int
+
+const (
+	RNConnectionType_Unknown  RNConnectionType = 0
+	RNConnectionType_Outbound RNConnectionType = 1
+	RNConnectionType_Inboound RNConnectionType = 2
+)
+
+type RNConnectionStatus int
+
+const (
+	RNConnectionStatus_NotConnected RNConnectionStatus = 0
+	RNConnectionStatus_Connected    RNConnectionStatus = 1
+	RNConnectionStatus_Attempted    RNConnectionStatus = 2
+	RNConnectionStatus_Terminated   RNConnectionStatus = 3
+)
+
+//type RemoteNodeId struct {
+//	PeerId    uint64
+//	AttemptId uint64
+//}
+
+type RemoteNode struct {
+	// Should we have ID here?
+	peer             *Peer
+	id               RemoteNodeId
+	connectionType   RNConnectionType
+	connectionStatus RNConnectionStatus
+
+	params *DeSoParams
+
+	bc  *Blockchain
+	srv *Server
```

Comment:
There is one place where we need to rely on the Server, which is to signal the PeerHandshakeComplete message. If we used ConnectionManager for this, it would basically just call on Server to signal this message anyway. Besides this one, I purged every other Server reference.